### PR TITLE
Fix PME direct-space and add host-guest overlap test

### LINKED ISSUE #45: Direct-space PME contribution incorrect if alpha_ewald = 0
If the reference `System` object does not manually specify `alpha_ewald` for PME, it appears that the direct-space PME contribution is incorrect. The logic in [this block](https://github.com/choderalab/alchemy/blob/master/alchemy/alchemy.py#L952-L958) does not work as intended, instead skipping the `if alpha_ewald == 0.0` clause (since `alpha_ewald` has units of 1/nm) and using a value of 0 in the calculation of the electrostatic energy expression.

If `alpha_ewald` is not manually set, it will be determined automatically from the PME tolerance upon `Context` creation. This is thankfully standardized in [`NonbondedForceImpl`](https://github.com/pandegroup/openmm/blob/786bc04f548074c1c936b4ec0a8a827fbadd7420/openmmapi/src/NonbondedForceImpl.cpp#L141-L142):

``` C++
    double tol = force.getEwaldErrorTolerance();
    alpha = (1.0/force.getCutoffDistance())*std::sqrt(-log(2.0*tol));
```

### PR DESCRIPTION
**Pull request by jchodera**
Fixes #45 and adds a host-guest test.

### CODE REVIEW

**Comment by jchodera** [comment]
This should be ready to review and merge.

Note that I think the overlap tests don't run on travis, so we'll want to enable jenkins testing

**Comment by jchodera** [comment]
Now I'm seeing a failure [here](https://travis-ci.org/choderalab/alchemy/jobs/165152252#L750-L759)---not sure which system yet:

```
======================================================================
ERROR: Compare annihilated electrostatics / decoupled sterics states in vacuum and a large box.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/miniconda/envs/test/lib/python2.7/site-packages/alchemy-1.3.0-py2.7.egg/alchemy/tests/test_alchemy.py", line 336, in test_denihilated_states
    raise Exception("Maximum allowable difference lambda=1 energy and lambda=0 energy in vacuum and periodic box exceeded (was %.8f kcal/mol; allowed %.8f kcal/mol); test failed." % (delta / unit.kilocalories_per_mole, MAX_DELTA / unit.kilocalories_per_mole))
Exception: Maximum allowable difference lambda=1 energy and lambda=0 energy in vacuum and periodic box exceeded (was 2.74731396 kcal/mol; allowed 0.59616198 kcal/mol); test failed.
-------------------- >> begin captured logging << --------------------
```

**Comment by jchodera** [comment]
It's not printing out the logger info before the exception is thrown---how can I fix this?
https://github.com/choderalab/alchemy/blob/5bea183a0a3a42a9746f9e2961f21492ee9f0920/alchemy/tests/test_alchemy.py#L326-L336

**Comment by jchodera** [comment]
I think the failure is likely due to the fact that PME for the ligand now is correctly tapered by `erfc(alpha*r)`, so that the fully-interacting `lambda=1` state for PME will differ from the `lambda=1` state in vacuum. I imagine we can just disable this test for now and think of a better test.

File: alchemy/tests/test_alchemy.py [hunk 1]
```diff
@@ -957,8 +967,10 @@ def overlap_check(reference_system, positions, platform_name=None, precision=Non
 
     # Raise an exception if the error is larger than 3kT.
     MAX_DEVIATION = 3.0 # kT
+    report = "DeltaF = %12.3f +- %12.3f kT (%5d samples, g = %6.1f); du mean %.3f kT stddev %.3f kT" % (DeltaF, dDeltaF, Neff, g, du_n.mean(), du_n.std())
+    logger.info(report)
+    print(report)
```
Comment (bas-rustenburg) [inline alchemy/tests/test_alchemy.py:972]: Should we remove the print statement here, if there is already a logger statement?

Comment (jchodera) [reply alchemy/tests/test_alchemy.py:972]: I wasn't able to get the logging to print, so I needed to insert that to figure out what the problem was.

**Comment by jchodera** [comment]
Can anyone suggest how I get the logging to print?

File: devtools/travis-ci/install.sh [hunk 2]
```diff
@@ -12,4 +12,4 @@ bash $MINICONDA -b -p $HOME/miniconda
 
 # Configure miniconda
 export PATH=$HOME/miniconda/bin:$PATH
-conda install --yes conda-build anaconda-client jinja2 pip
+conda install --yes conda-build==2.0.1 anaconda-client jinja2 pip
```
Comment (bas-rustenburg) [inline devtools/travis-ci/install.sh:15]: I think the syntax contains only a single `=` sign.

File: alchemy/tests/test_alchemy.py (hunk 1 above)
Comment (bas-rustenburg) [reply alchemy/tests/test_alchemy.py:972]: Got it. It can be a bit of a pain to handle logger objects between modules/files.

**Comment by jchodera** [comment]
I've temporarily disabled the test that was failing---which likely can no longer actually work with the PME fix. We can figure out how to fix both (1) logging and (2) that test in a separate PR.

**Comment by jchodera** [comment]
Ready for review by @andrrizzi (or @Lnaden) and merge!

**Comment by jchodera** [comment]
This should hopefully fix the PME overlap issues, or at least greatly reduce them.

**Comment by andrrizzi** [comment]
Awesome! Thanks for catching this. Looks great to me.

**Comment by jchodera** [comment]
Merged.

Should we cut a new bugfix release?

**Comment by Lnaden** [comment]
Yes, since this does fix a major bug in the alchemical PME treatment

**Pull request merged**